src/components/app-switcher/app-switcher.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import Icon from 'components/icon';
+
+/**
+ * Renders an application (drawer) switcher
+ * @class AppSwitcher
+ */
+
+export default class AppSwitcher extends React.Component {
+
+  static propTypes = {
+        /**
+         * The json blob for the application links
+         *
+         * @property applicationJson
+         * @type {String}
+         * @default ''
+         */
+    applicationJson: PropTypes.string.isRequired,
+
+        /**
+         * The title of the menu item
+         *
+         * @property menuTitle
+         * @type {String}
+         * @default ''
+         */
+    menuTitle: PropTypes.string
+
+  }
+
+    /**
+     * Pseudo private class for the styles
+     *
+     * @method styles
+     * @private
+     * @return Hash
+     */
+
+  styles = () => {
+    return {
+      appItem: "carbon-switcher-app-item",
+      menu: classNames((this.state.active ? "carbon-switcher-menu-active" : ""), "carbon-switcher-menu-title"),
+      sectionTitle: "carbon-switcher-section-title",
+      drawer: "carbon-switcher-drop-down",
+      arrowUp: "carbon-switcher-arrow-up"
+    };
+  }
+
+    /**
+     * Constructor method setting properties
+     *
+     * @method constructor
+     * @return {Void}
+     */
+
+  constructor(...props) {
+    super(...props);
+
+    this.state = {
+      active: false
+    };
+  }
+
+    /**
+     * Handle when the user clicks on the main menu icon
+     *
+     * @method handleOpenWindow
+     * @return {Void}
+     */
+
+  handleOpenWindow = () => {
+    document.addEventListener('click', this.handleCloseWindow);
+    this.setState({active: true});
+  }
+
+    /**
+     * Handle when the user clicks anywhere on the screen after
+     * the main menu is activated
+     *
+     * @method handleCloseWindow
+     * @return {Void}
+     */
+
+  handleCloseWindow = () => {
+    document.removeEventListener('click', this.handleCloseWindow);
+    this.setState({active: false});
+  }
+
+    /**
+     * Compose each of the individual pieces from applicationJson
+     *
+     * @method composeDrawer
+     * @return JSX Elements
+     */
+
+  compose = () => {
+    let parsedJson = JSON.parse(this.props.applicationJson) || {};
```

Comment:
Normally we'd ensure it was JSON before passing it into the component (either within the Flux Store or when Rails renders the data to the page). Its a bit strange having the component request the prop as a string only for us to convert it to JSON.